### PR TITLE
Add rerun visualization

### PR DESCRIPTION
**Pull request by roym899**
Currently we can visualize the output at various stages of the optimization.
1. Input
2. (Paper Section 3.1) Initial optimization of human orientation and translation (alignment of PHALP output with ViTPose output)
3. (Paper Section 3.2) Optimization of camera scale, human shape, and body pose
4. (Paper Section 3.3) Optimization of camera scale, human motion priors (HuMoR), ground plane + ground contact -> Final output

The different phases are logged to separate timelines, and to a common timeline. This way we can switch between the results just by changing the timeline, but also compare the phases with multiple spatial views at the same time.

Final output and comparison of different stages

https://user-images.githubusercontent.com/9785832/226495966-05b1fc24-dabf-443f-905d-ed6fb07287a5.mp4

Switching with the timeline

https://user-images.githubusercontent.com/9785832/226495918-4b797842-be3b-4f64-9872-e6407d3059e9.mp4

An alternative would be to log to one timeline one after another. Ideal solution would be a separate optimization_step sequence timeline. These are just the final result from the stage, it would be nice if we could log the intermediate steps as well.

Some things that could be added:
- visualize ground plane (which is also estimated during the last optimization step)
- visualize camera trajectory (ideally this could be done in the viewer with stride + history)
- logging at optimization time instead of from results saved to disk (currently we are following the same approach as the original repo, replacing the visualization script with a rerun-based visualization script)

### CODE REVIEW

File: slahmr/run_rerun_vis.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+            xyz="RDF",
+        )
+        for i, _ in enumerate(dataset.track_ids):
+            if vis_mask[i][frame_id] >= 0:
+                rr.log_mesh(
+                    f"world/{phase_label}/#{i}",
+                    vertices[i, frame_id],
+                    indices=faces,
+                    normals=vertex_normals[i, frame_id],
+                    albedo_factor=_index_to_color(i),
```
Comment (nikolausWest) [inline slahmr/run_rerun_vis.py:163]: Could open an issue here for letting `AnnotationContext` apply to meshes as well

File: slahmr/run_rerun_vis.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+            [1, 2],
+            [0, 1],
+            [0, 2],
+            [1, 3],
+            [2, 4],
+            [3, 5],
+            [4, 6],
+        ]
+    )
+    idcs = [0, 16, 15, 18, 17, 5, 2, 6, 3, 7, 4, 12, 9, 13, 10, 14, 11]
```
Comment (nikolausWest) [inline slahmr/run_rerun_vis.py]: Break out as constants. Also, where do they come from?

File: slahmr/run_rerun_vis.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+        torch.multiprocessing.set_start_method("spawn")
+
+        with Pool(processes=len(args.gpus)) as pool:
+            res = pool.starmap(
+                launch_rerun_vis, [(i, args) for i in range(len(args.log_dirs))]
+            )
+        return
+
+    for i in range(len(args.log_dirs)):
+        launch_rerun_vis(i, args)
```
Comment (nikolausWest) [inline slahmr/run_rerun_vis.py]: I think there are a couple to many layers to the whole "launching rerun" stuff. I may be wrong but I believe it should be possible to simplify it quite some.

File: slahmr/run_rerun_vis.py (hunk 1 above)
Comment (roym899) [reply slahmr/run_rerun_vis.py:163]: will do

File: slahmr/run_rerun_vis.py (hunk 2 above)
Comment (roym899) [reply slahmr/run_rerun_vis.py]: Added some comments, and capitalized.

File: slahmr/run_rerun_vis.py (hunk 3 above)
Comment (roym899) [reply slahmr/run_rerun_vis.py]: Changed it to just two functions `main` which parses the arguments and prepares the dataset and `log_to_rerun`.

**Review by nikolausWest — APPROVED** [review]
Super nice!

**Pull request merged**